UnitTests/CardPaymentsTests/ConfirmPaymentSourceRequest_Tests.swift

```diff
@@ -15,38 +15,170 @@ class ConfirmPaymentSourceRequest_Tests: XCTestCase {
-                            "attributes": {
-                                "verification": {
-                                    "method": "SCA_WHEN_REQUIRED"
-                                }
-                        },
-                            "expiry": "2024-11"
-                        }
-                    }
-                }
-                """.replacingOccurrences(of: " ", with: "").replacingOccurrences(of: "\n", with: "")
-            
-            XCTAssertEqual(paymentSourceBodyString, expectedPaymentSourceBodyString)
-        }
+        let expectedPaymentSourceDict: [String: Any] = [
+            "application_context": [
+                "return_url": "sdk.ios.paypal://card/success",
+                "cancel_url": "sdk.ios.paypal://card/cancel"
+            ],
+            "payment_source": [
+                "card": [
+                    "number": "4032036247327321",
+                    "security_code": "222",
+                    "billing_address": NSNull(),
+                    "name": NSNull(),
+                    "attributes": [
+                        "vault": [
+                            "store_in_vault": "ON_SUCCESS"
+                        ],
+                        "verification": [
+                            "method": "SCA_WHEN_REQUIRED"
+                        ],
+                        "customer": [
+                            "id": "testCustomer1"
+                        ]
+                    ],
+                    "expiry": "2024-11"
+                ] as [String: Any]
+            ]
+        ]
+        let paymentSourceDict = try JSONSerialization.jsonObject(with: paymentSourceBody, options: []) as! [String: Any]
+        XCTAssertEqual(paymentSourceDict as NSDictionary, expectedPaymentSourceDict as NSDictionary)
+    }
+    
+    func testEncodingPaymentSource_withValidCard_andVaultWithPurchase_noCustomerID() throws {
+        let mockOrderID = "mockOrderID"
+        let card = Card(
+            number: "4032036247327321",
+            expirationMonth: "11",
+            expirationYear: "2024",
+            securityCode: "222"
+        )
+
+        let vault = Vault()
+        let cardRequest = CardRequest(orderID: mockOrderID, card: card, vault: vault)
+
+        let confirmPaymentSourceRequest = try XCTUnwrap(
+            ConfirmPaymentSourceRequest(clientID: "fake-token", cardRequest: cardRequest)
+        )
+
+        let paymentSourceBody = try XCTUnwrap(confirmPaymentSourceRequest.body)
+        let expectedPaymentSourceDict: [String: Any] = [
+            "application_context": [
+                "return_url": "sdk.ios.paypal://card/success",
+                "cancel_url": "sdk.ios.paypal://card/cancel"
+            ],
+            "payment_source": [
+                "card": [
+                    "number": "4032036247327321",
+                    "security_code": "222",
+                    "billing_address": NSNull(),
+                    "name": NSNull(),
```

Comment:
Is there a reason we need to use `NSNull` here instead of `nil`?

Reply:
Because I am converting it to NSDictionary to be able to do equality on nested dictionary, I thought I needed to use NSNull(). I think I got syntax error trying it with `nil` initially.

Reply:
addressed in commit 552e6ef914120af69d81f61a2c023fe85a62f809